frontend/src/metabase/admin/tasks/containers/TaskModal/TaskModal.tsx

```diff
@@ -0,0 +1,90 @@
+import { useMemo } from "react";
+import { Link } from "react-router";
+import { goBack } from "react-router-redux";
+import { t } from "ttag";
+
+import { useGetTaskQuery } from "metabase/api";
+import { CodeBlock } from "metabase/components/CodeBlock";
+import { CopyButton } from "metabase/components/CopyButton";
+import { LoadingAndErrorWrapper } from "metabase/components/LoadingAndErrorWrapper";
+import ModalContent from "metabase/components/ModalContent";
+import { openSaveDialog } from "metabase/lib/dom";
+import { useDispatch } from "metabase/lib/redux";
+import { Box, Button, Flex, Icon } from "metabase/ui";
+import type { Task } from "metabase-types/api";
+
+import S from "./TaskModal.module.css";
+
+interface Props {
+  params: { taskId: number };
+}
+
+export const TaskModal = ({ params }: Props) => {
+  const dispatch = useDispatch();
+  const { data: task, error, isLoading } = useGetTaskQuery(params.taskId);
+  const code = useMemo(() => formatTaskDetails(task), [task]);
```

Comment:
Nit: not sure this needs a `useMemo`, `JSON.stringify` is pretty fast and the resulting string should be stable.

Reply:
e2cf880d817bf2cda743fa97520f25fc953cd24

I still kept useMemo for the `code.split('\n')` bit.